src/ray/protobuf/autoscaler_state_service.proto

```diff
@@ -0,0 +1,123 @@
+syntax = "proto3";
+option cc_enable_arenas = true;
+
+package ray.rpc;
+
+import "src/ray/protobuf/intance_manager.proto";
+// ============= Cluster Resources ====================
+//
+// Following fields represents the Cluster Resources autoscaler interested
+// in.
+
+// Represents an anti-affinity constraint. A bundle with this constraint
+// can't be allocated to a node that has a label with the same name and
+// value.
+message AntiAffinityConstraint {
+  string label_name = 1;
+  string label_value = 2;
+}
+
+message PlacementConstraint {
+  oneof constraint {
+    AntiAffinityConstraint anti_affinity = 1;
+  }
+}
+
+message ResourceBundle {
+  map<string, double> resources = 1;
+  PlacementConstraint placement_constraint = 2;
+  // If the resource bundle is allocated, the node will be labeled
+  // with the following labels.
+  // map<string, string> new_labels = 3;
+}
+
+// All bundles in the same resource request require gang
+// allocation semantics: they should be allocated together.
+message GangSchedulingRequest {
+  // a map from bundles to the number of bundles requested.
+  map<ResourceBundle, int32_t> bundles = 1;
+}
+
+// Cluster constraint represents minimial cluster size requirement,
+// issued through ray.autoscaler.sdk.request_resources.
+message OnDemandResourceRequest {
+  bytes request_id = 1;
```

Comment:
what's the point of this id field? imo we shouldn't add it unless we know we need it (we shouldn't future proof these things because there is no cost to adding the field later).